core/src/OpenMP/Kokkos_OpenMP_Instance.hpp

```diff
@@ -73,10 +73,13 @@ class OpenMPInternal {
 
   HostThreadTeamData* m_pool[OpenMPTraits::MAX_THREAD_COUNT];
 
+  static OpenMPInternal* s_singleton;
+
  public:
   friend class Kokkos::OpenMP;
 
   static OpenMPInternal& singleton();
+  static void reset_singleton();
```

Comment:
hmm ... wouldn't reset imply that we can reuse after? maybe both should be called `allocate/delete_singleton` respectively

Reply:
Do you mean like now ? :smiley:

Reply:
Outdated.